gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/VersionProps.flex

```diff
@@ -23,52 +23,52 @@ import com.intellij.psi.TokenType;
 
 %%
 
+// Define the lexer class
 %class VersionPropsLexer
 %implements FlexLexer
 %unicode
 %function advance
 %type IElementType
-%eof{  return;
+%eof{ return; }
 %eof}
 
+// Define lexer states
+%state WAITING_NAME, WAITING_VERSION, WAITING_VALUE, WAITING_COMMENT, INVALID_VALUE
+
+// Define token patterns
 CRLF=\R
 WHITE_SPACE=[\ \n\t\f]
-VALUE_CHARACTER=[^ \n\f]+
+VALUE=[^ \n\f#]+
 COLON=[:]
 EQUALS=[=]
 DOT=[.]
 KEY_CHARACTER=[^:=\ \n\t\f]+
 GROUP_PART=[^.:=\ \n\t\f]+
 COMMENT=("#")[^\r\n]*
 
-%state WAITING_NAME, WAITING_VERSION, WAITING_VALUE, INVALID_VALUE
-
 %%
-<YYINITIAL> {COMMENT}                                       { yybegin(YYINITIAL); return VersionPropsTypes.COMMENT; }
-
-<YYINITIAL> {GROUP_PART}+                                   { yybegin(YYINITIAL); return VersionPropsTypes.GROUP_PART; }
-<YYINITIAL> {DOT}                                           { yybegin(YYINITIAL); return VersionPropsTypes.DOT; }
-
-<YYINITIAL> {COLON}                                         { yybegin(WAITING_NAME); return VersionPropsTypes.COLON; }
-
-<WAITING_NAME> {KEY_CHARACTER}+                             { yybegin(WAITING_VERSION); return VersionPropsTypes.NAME_KEY; }
-
-<WAITING_NAME> {WHITE_SPACE}+                               { yybegin(WAITING_NAME); return TokenType.WHITE_SPACE; }
-
-<WAITING_NAME> {CRLF}({CRLF}|{WHITE_SPACE})+                { yybegin(YYINITIAL); return TokenType.WHITE_SPACE; }
 
-<WAITING_VERSION> {EQUALS}                                  { yybegin(WAITING_VALUE); return VersionPropsTypes.EQUALS; }
+<YYINITIAL> {COMMENT}                            { yybegin(YYINITIAL); return VersionPropsTypes.COMMENT; }
+<YYINITIAL> {GROUP_PART}                         { yybegin(YYINITIAL); return VersionPropsTypes.GROUP_PART; }
+<YYINITIAL> {DOT}                                { yybegin(YYINITIAL); return VersionPropsTypes.DOT; }
+<YYINITIAL> {WHITE_SPACE}*{DOT}{WHITE_SPACE}*    { yybegin(INVALID_VALUE); return TokenType.BAD_CHARACTER; }
+<YYINITIAL> {COLON}                              { yybegin(WAITING_NAME); return VersionPropsTypes.COLON; }
+<YYINITIAL> {WHITE_SPACE}*{COLON}{WHITE_SPACE}*  { yybegin(INVALID_VALUE); return TokenType.BAD_CHARACTER; }
 
-<WAITING_VERSION> {WHITE_SPACE}+                            { yybegin(WAITING_VERSION); return TokenType.WHITE_SPACE; }
+<WAITING_NAME> {KEY_CHARACTER}                   { yybegin(WAITING_VERSION); return VersionPropsTypes.NAME_KEY; }
```

Comment:
nit - maybe change this to `KEY` (as it can contain multiple characters)

Reply:
done